buffer/src/main/java/io/netty/buffer/SimpleLeakAwareByteBuf.java

```diff
@@ -41,20 +43,28 @@ public ByteBuf touch(Object hint) {
 
     @Override
     public boolean release() {
-        boolean deallocated =  super.release();
-        if (deallocated) {
-            leak.close();
+        // Call unwrap() before just in case that super.release() will change the ByteBuf instance that is returned
+        // by ByteBuf.
+        ByteBuf unwrapped = unwrap();
+        if (super.release()) {
+            boolean closed = ResourceLeakDetector.close(leak, unwrapped);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "closed". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)